src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -30,6 +30,8 @@ class Definition
     private $factoryMethod;
     private $factoryService;
     private $shared = true;
+    private $deprecated = false;
+    private $deprecationTemplate = "The '%service_id%' service is deprecated. You should stop using it, as it will soon be removed.";
```

Comment:
I meant: around the service id, see e.g. https://github.com/symfony/symfony/blob/2.8/src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php#L379